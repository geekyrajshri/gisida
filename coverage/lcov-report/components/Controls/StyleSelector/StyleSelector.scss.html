<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Controls/StyleSelector/StyleSelector.scss</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">components/Controls/StyleSelector</a> StyleSelector.scss
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">/* Borrowed from Leaflet layer control styles */
&nbsp;
/* control positioning */
.leaflet-control {
	position: relative;
	z-index: 7;
	pointer-events: auto;
	}
.leaflet-top,
.leaflet-bottom {
	position: absolute;
	z-index: 1000;
	pointer-events: none;
	}
.leaflet-top {
	top: 0;
	}
.leaflet-right {
	right: 0;
	}
.leaflet-bottom {
	bottom: 0;
	}
.leaflet-left {
	left: 0;
	}
.leaflet-control {
	float: left;
	clear: both;
	}
.leaflet-right .leaflet-control {
	float: right;
	}
.leaflet-top .leaflet-control {
	margin-top: 10px;
	}
.leaflet-bottom .leaflet-control {
	margin-bottom: 10px;
	}
.leaflet-left .leaflet-control {
	margin-left: 10px;
	}
.leaflet-right .leaflet-control {
	margin-right: 10px;
	}
&nbsp;
/* layers control */
.leaflet-control-layers-toggle {
	background-position: 50% 50%;
	background-repeat: no-repeat;
	display: block;
	}
.leaflet-control-layers {
	box-shadow: 0 1px 5px rgba(0,0,0,0.4);
	background: #fff;
	border-radius: 5px;
	}
.leaflet-control-layers-toggle {
	background-image: url(/assets/img/layers.png);
	width: 32px;
	height: 32px;
	}
.leaflet-retina .leaflet-control-layers-toggle {
	background-image: url(/assets/img/layers-2x.png);
	background-size: 26px 26px;
	}
.leaflet-touch .leaflet-control-layers-toggle {
	width: 44px;
	height: 44px;
	}
.leaflet-control-layers .leaflet-control-layers-list,
.leaflet-control-layers:hover .leaflet-control-layers-toggle {
	display: none;
	}
.leaflet-control-layers:hover .leaflet-control-layers-list {
	display: block;
	position: relative;
	}
.leaflet-control-layers:hover{
	padding: 6px 10px 6px 6px;
	color: #333;
	background: #fff;
	}
.leaflet-control-layers-selector {
	margin-top: 2px;
	position: relative;
	top: 1px;
	}
.leaflet-control-layers label {
	display: block;
	}
.leaflet-control-layers-separator {
	height: 0;
	border-top: 1px solid #ddd;
	margin: 5px -10px 5px -6px;
	}
&nbsp;
.layer-selector {
  margin-top: 100px; 
}
.leaflet-control-layers-base label input{
  margin-right: 5px;
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jun 27 2017 15:17:21 GMT+0300 (EAT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
