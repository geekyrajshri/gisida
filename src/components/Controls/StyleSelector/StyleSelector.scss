/* Borrowed from Leaflet layer control styles */

/* control positioning */
.leaflet-control {
  position: relative;
  z-index: 7;
  pointer-events: auto;
}

.leaflet-top,
.leaflet-bottom {
  position: absolute;
  z-index: 1000;
  pointer-events: none;
}

.leaflet-top {
  top: 0;
}

.leaflet-right {
  right: 0;
}

.leaflet-bottom {
  bottom: 0;
}

.leaflet-left {
  left: 0;
}

.leaflet-control {
  float: left;
  clear: both;
}

.leaflet-right .leaflet-control {
  float: right;
}

.leaflet-top .leaflet-control {
  margin-top: 10px;
}

.leaflet-bottom .leaflet-control {
  margin-bottom: 10px;
}

.leaflet-left .leaflet-control {
  margin-left: 10px;
}

.leaflet-right .leaflet-control {
  margin-right: 10px;
}

/* layers control */
.leaflet-control-layers-toggle {
  background-position: 50% 50%;
  background-repeat: no-repeat;
  display: block;
}

.leaflet-control-layers {
  box-shadow: 0 1px 5px rgba(0, 0, 0, 0.4);
  background: #fff;
  border-radius: 5px;
}

.leaflet-control-layers-toggle {
  background-image: url(/assets/img/layers.png);
  width: 32px;
  height: 32px;
}

.leaflet-retina .leaflet-control-layers-toggle {
  background-image: url(/assets/img/layers-2x.png);
  background-size: 26px 26px;
}

.leaflet-touch .leaflet-control-layers-toggle {
  width: 44px;
  height: 44px;
}

.leaflet-control-layers .leaflet-control-layers-list,
.leaflet-control-layers:hover .leaflet-control-layers-toggle {
  display: none;
}

.leaflet-control-layers:hover .leaflet-control-layers-list {
  display: block;
  position: relative;
}

.leaflet-control-layers:hover {
  padding: 6px 10px 6px 6px;
  color: #333;
  background: #fff;
}

.leaflet-control-layers-selector {
  margin-top: 2px;
  position: relative;
  top: 1px;
}

.leaflet-control-layers label {
  display: block;
}

.leaflet-control-layers-separator {
  height: 0;
  border-top: 1px solid #ddd;
  margin: 5px -10px 5px -6px;
}

.layer-selector {
  margin-top: 100px;
}

.leaflet-control-layers-base label input {
  margin-right: 5px;
}
