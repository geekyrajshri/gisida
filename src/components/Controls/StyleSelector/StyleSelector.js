require('./StyleSelector.scss');

function StyleSelector(props) {
  return (
    <div className="leaflet-left leaflet-top leaflet-right layer-selector">
      <div aria-haspopup="true" className="leaflet-control leaflet-control-layers">
        <a title="styles" className="leaflet-control-layers-toggle"></a>
        <form className="leaflet-control-layers-list">
          <div className="leaflet-control-layers-base">
            {props.styles.map((b, i) =>
              <label key={"label_" + i}>
                <input
                  readOnly
                  key={"input_" + i}
                  type="radio" name="leaflet-base-layers" className="leaflet-control-layers-selector"
                  value={b.style}
                  onClick={(e) => props.changeStyle(e.target.value)}
                  checked={props.style === b.style}></input>
                <span>{b.label}</span>
              </label>)
            }
          </div>
          <div className="leaflet-control-layers-overlays"></div>
        </form>
      </div>
    </div>
  );
}

export default StyleSelector
