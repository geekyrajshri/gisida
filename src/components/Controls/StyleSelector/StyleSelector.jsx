import React from 'react';

require('./StyleSelector.scss');

function StyleSelector(props) {
  return (
    <div className="leaflet-left leaflet-top leaflet-right layer-selector">
      <div aria-haspopup="true" className="leaflet-control leaflet-control-layers">
        <a title="styles" className="leaflet-control-layers-toggle" />
        <form className="leaflet-control-layers-list">
          <div className="leaflet-control-layers-base">
            {props.styles.map(b =>
              (<label key={`label_${b.label}`} htmlFor="styles">
                <input
                  readOnly
                  key={`input_${b.label}`}
                  type="radio"
                  name="leaflet-base-layers"
                  className="leaflet-control-layers-selector"
                  value={b.style}
                  onClick={e => props.changeStyle(e.target.value)}
                  checked={props.style === b.style}
                />
                <span>{b.label}</span>
              </label>))
            }
          </div>
          <div className="leaflet-control-layers-overlays" />
        </form>
      </div>
    </div>
  );
}

StyleSelector.propTypes = {
  styles: React.PropTypes.objectOf(React.PropTypes.any).isRequired,
  style: React.PropTypes.string.isRequired,
};

export default StyleSelector;
