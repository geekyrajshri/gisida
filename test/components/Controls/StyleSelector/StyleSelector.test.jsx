import React from 'react';
import renderer from 'react-test-renderer';
import StyleSelector from '../../../../src/components/Controls/StyleSelector/StyleSelector';
import styles from '../../../fixtures/styles.json';

describe('StyleSelector', () => {
    it('StyleSelector renders map style icon and options', () => {
        const changeStyle = jest.fn();
        const style = 'test://test-style';
        const component = renderer.create(
            <StyleSelector style={style} styles={styles} />);
        const json = component.toJSON();
        expect(json).toMatchSnapshot();
    });
});
