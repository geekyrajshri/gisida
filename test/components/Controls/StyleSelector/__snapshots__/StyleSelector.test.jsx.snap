// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleSelector StyleSelector renders map style icon and options 1`] = `
<div
  className="leaflet-left leaflet-top leaflet-right layer-selector"
>
  <div
    aria-haspopup="true"
    className="leaflet-control leaflet-control-layers"
  >
    <a
      className="leaflet-control-layers-toggle"
      title="styles"
    />
    <form
      className="leaflet-control-layers-list"
    >
      <div
        className="leaflet-control-layers-base"
      >
        <label
          htmlFor="styles"
        >
          <input
            checked={false}
            className="leaflet-control-layers-selector"
            name="leaflet-base-layers"
            onClick={[Function]}
            readOnly={true}
            type="radio"
            value="test://test-style-1"
          />
          <span>
            style-1
          </span>
        </label>
        <label
          htmlFor="styles"
        >
          <input
            checked={false}
            className="leaflet-control-layers-selector"
            name="leaflet-base-layers"
            onClick={[Function]}
            readOnly={true}
            type="radio"
            value="test://test-style-2"
          />
          <span>
            style-2
          </span>
        </label>
        <label
          htmlFor="styles"
        >
          <input
            checked={false}
            className="leaflet-control-layers-selector"
            name="leaflet-base-layers"
            onClick={[Function]}
            readOnly={true}
            type="radio"
            value="test://test-style-3"
          />
          <span>
            style-3
          </span>
        </label>
      </div>
      <div
        className="leaflet-control-layers-overlays"
      />
    </form>
  </div>
</div>
`;
